data/human/jobs.txt

```diff
@@ -1550,6 +1550,23 @@ mission "Cargo [4]"
 		payment 4000
 		dialog phrase "generic cargo delivery payment"
 
+mission "Cargo [5]"
+	name "Delivery to <planet>"
+	job
+	repeat
+	description "Deliver <cargo> to <destination>. Payment is <payment>."
+	cargo random 5 2 .1
+	source
+		government "Republic" "Free Worlds" "Syndicate" "Neutral" "Independent"
+	destination
+		distance 0 3
```

Comment:
True, but it also opens them up for in-system deliveries, such as Earth-Luna. Ideally, I'd like a restrictor on there that blocks same-planet destinations while still allowing same-system deliveries; but well, while expensive, I'd expect hiring a spaceship to transport one around a planet *is* a valid transportation option...

Reply:
> while expensive, I'd expect hiring a spaceship to transport one around a planet is a valid transportation option...

Given that ES doesn't present planets as having many different places to land outside of missions with their own unique conversations or dialogs, this is not how it will look to most people: it'll look like low quality work.

Until we have a way to specifically prevent choosing the source as the destination, you shouldn't do this.

Even when we do add such a feature, it will likely require its own keyword, not be automatically applied to all destination selection processes (a person may specifically want the source to be an option for some other reason) so this will need to be changed at that point anyway.

Reply:
Definitely. It shouldn't be a baked-in restriction, but rather some kind of a term that can be added to the filter, just like any of the myriad other options we have there, and for specifically that reason. I already have an issue open requesting such a term: https://github.com/endless-sky/endless-sky/issues/7789

But sure, I could swap these to be `1 3` until such a term is implemented. Kind of frustrating, since I intentionally wanted it to be possible to have running around Sol to be a valid activity, but that can wait a bit longer, I suppose.

Reply:
Actually, it *would* be possible to implement a filter along the lines of:

or:
- Sol and distance 0 3
- distance 1 3
But even in Sol, that'd still be a 1 in 3 chance of picking the starting planet as destination.

Reply:
In the new Hai Revealed text there's dialogue about shuttles being used to ferry goods planetside, which makes sense because Earth is so big and there's going to be a lot of demand. It might still work. And anyway, why can't you use your starship to move people / goods around on the same planet, or stay more than one day on a planet?

Reply:
Looks like this set of missions is kinda blocked on 'not source', i.e. #7789 mentioned above?

Reply:
No, it's not blocked. #7789 would allow it to be modified to enable working on intra-system opportunities too, but that's an edge benefit to the primary goal of enabling short-haul jobs that the starter ships are capable of doing; and that goal can be handled perfectly fine under current mechanics.

Reply:
Something to remember when it is possible to have the job point to another planet in the same system without also considering the source as an option, I think the default `deadline` is 2x the number of jumps from source to destination. With in system destinations, this would result in a deadline of 0 days from when the job is offered, which is a problem.

Reply:
But if we had a single mission set that was only source 0, destination 0, we wouldn't need to make these rush jobs with deadlines?

Reply:
I'm sorry, I think I'm missing something there. What are you trying to do?

Reply:
This PR right now is for a mission set with range of 1 to 3 systems away from the source system. How about another mission set with only a range of 0, where we can avoid those problems, and have the dialogue work? It would be like "Transport these goods to a sister settlement on Mars" or "A city on the other side of Earth needs an emergency supply of food to meet demand. Transport this food from the main spaceport to the city's closest regional spaceport."